src/pages/Task.tsx

```diff
@@ -76,7 +76,11 @@ export const Task: React.FC = () => {
   const isPatchOnCommitQueue = requester === commitQueueRequester;
 
   // Set the execution if it isnt provided
-  if (Number.isNaN(selectedExecution) && latestExecution !== undefined) {
+  if (
+    Number.isNaN(selectedExecution) &&
+    task?.id === id &&
```

Comment:
Ah okay. So this was something I was encountering when a task had multiple executions, and this was the only way I could think to fix it (at least quickly, haha).

When a task has multiple executions, and you click to go to the base task, for some reason the previous task's data will initially persist. You can see in the video attached that the task data's `id` doesn't match with the URL `id`. Since we base the `latestExecution` field off of the task data, that means it tries to go to execution 2 for the base task, which doesn't exist.

https://user-images.githubusercontent.com/47064971/153973587-02f74d57-6262-4f7f-9af6-1f874134d1eb.mov

This actually happens for any task, but most tasks just have 1 execution, so most of the time it doesn't cause an error. I don't really know why this data persists and couldn't find a way to "clear" it :(

Reply:
nvm it should be fixed now!!! just needed a `useEffect` :P